src/test/java/de/tum/in/www1/artemis/course/CourseTestService.java

```diff
@@ -1914,6 +1924,128 @@ public Course testArchiveCourseWithTestModelingAndFileUploadExercises() throws E
+        try (MockedStatic<ImageIO> mockedImageIO = mockStatic(ImageIO.class)) {
+            mockedImageIO.when(() -> ImageIO.read(any(File.class))).thenThrow(new IOException());
+            var archivePath = courseExamExportService.exportCourse(course, courseArchivesDirPath, exportErrors);
+            assertThat(archivePath).isNotEmpty();
+            Predicate<Path> missingPathPredicate = path -> path.getFileName().toString().contains("dragAndDropQuestion") && path.getFileName().toString().endsWith(".pdf");
+            extractAndAssertMissingContent(archivePath.orElseThrow(), quizSubmission, missingPathPredicate);
+        }
+    }
+
+    public void testArchiveCourseWithQuizExerciseCannotCreateParticipationDirectory() throws IOException {
+        List<String> exportErrors = Collections.synchronizedList(new ArrayList<>());
+        var course = courseUtilService.createCourse();
+        quizExerciseUtilService.addQuizExerciseToCourseWithParticipationAndSubmissionForUser(course, userPrefix + "student1", false);
+        try (MockedStatic<DataExportUtil> mockedFiles = mockStatic(DataExportUtil.class)) {
+            mockedFiles.when(() -> DataExportUtil.createDirectoryIfNotExistent(any())).thenThrow(new IOException());
+            var archivePath = courseExamExportService.exportCourse(course, courseArchivesDirPath, exportErrors);
+            assertThat(archivePath).isNotEmpty();
+        }
+
+    }
+
+    public void testArchiveCourseWithQuizExerciseCannotExportMCAnswersSubmission() throws IOException {
+        List<String> exportErrors = Collections.synchronizedList(new ArrayList<>());
+        var course = courseUtilService.createCourse();
+        var quizSubmission = quizExerciseUtilService.addQuizExerciseToCourseWithParticipationAndSubmissionForUser(course, userPrefix + "student1", false);
+        try (MockedStatic<org.apache.commons.io.FileUtils> mockedFiles = mockStatic(org.apache.commons.io.FileUtils.class)) {
+            mockedFiles
+                    .when(() -> org.apache.commons.io.FileUtils.writeLines(argThat(file -> file.toString().contains("multiple_choice_questions_answers")), anyString(), anyList()))
+                    .thenThrow(new IOException());
+            var archivePath = courseExamExportService.exportCourse(course, courseArchivesDirPath, exportErrors);
+            assertThat(archivePath).isNotEmpty();
+            Predicate<Path> missingPathPredicate = path -> path.getFileName().toString().contains("multiple_choice_questions_answers")
+                    && path.getFileName().toString().endsWith(".txt");
+            extractAndAssertMissingContent(archivePath.orElseThrow(), quizSubmission, missingPathPredicate);
+            assertThat(exportErrors).hasSize(1);
+            assertThat(exportErrors.get(0)).contains("Failed to export multiple choice answers");
+        }
+    }
+
+    public void testArchiveCourseWithQuizExerciseCannotExportSAAnswersSubmission() throws IOException {
```

Comment:
Could you use parameterized tests? Teamscale complains about code duplication and I kind of agree.